### PR TITLE
Do not permit to use ExpandExpressions on expressions, that can't be expanded

### PR DESCRIPTION
**Pull request by timramone**

### CODE REVIEW

**Comment by timramone** [comment]
@IharBury check this out please when have time

**Comment by timramone** [comment]
Maybe someone else could rewiev this? 
@mijkathegreat @gnhdnb could you please take a look?

File: Mindbox.Expressions/ExpressionExpander.cs [hunk 1]
```diff
@@ -210,47 +215,42 @@ protected override Expression VisitUnary(UnaryExpression node)
 			return baseResult;
 		}
 
-
-		private bool TrySubstituteExpression(
+		private Expression SubstituteExpression(
 			Expression expressionExpression, 
 #if NET45 || CORE45 || WINDOWS_PHONE_APP
 			IReadOnlyList<Expression> arguments,
```
Comment (mijkathegreat) [inline Mindbox.Expressions/ExpressionExpander.cs]: Wouldn't compile under NET45 || CORE45 || WindowsPhone cause of comma at the end :)
Maybe we can remove all the conditional compilation?

File: Mindbox.Expressions/ExpressionExpander.cs [hunk 2]
```diff
@@ -166,11 +167,13 @@ protected override Expression VisitMethodCall(MethodCallExpression node)
 				var constantExpression = (ConstantExpression)baseResult.Object;
 				if (IsEvaluateMethod((MethodInfo)constantExpression.Value))
 				{
-					var innerExpression = TryGetLambdaExpressionFromExpression(baseResult.Arguments[1]);
-					if (innerExpression != null)
-						return Visit(ExpressionParameterSubstitutor.SubstituteParameters(
-							innerExpression,
-							new Dictionary<ParameterExpression, Expression>()));
+					var innerExpression = GetLambdaExpressionFromExpression(baseResult.Arguments[1]);
+					if (innerExpression == null)
```
Comment (gnhdnb) [inline Mindbox.Expressions/ExpressionExpander.cs]: Is the null check really needed after every call of `GetLambdaExpressionFromExpression`?

Comment (timramone) [reply Mindbox.Expressions/ExpressionExpander.cs]: It's not :)

**Pull request merged**